lib/oauth2-client.js

```diff
@@ -22,68 +23,41 @@ bedrock.events.on('bedrock.init', async () => {
   });
 });
 
-export class OAuth2Client {
-  /**
-   * Constructs an OAuth2Client instance to return an access token. It first
-   * checks the database to see if an access token already exists for the client
-   * id; if not, it creates an access token then stores it in the database.
-   *
-   * @param {object} options - Options to use.
-   * @param {string} options.config - The "oauth2-client-grant" properties.
-   * @param {object} options.httpsAgent - A NodeJS HTTPS Agent (`https.Agent`).
-   * @param {string} options.issuer - The issuer.
-   * @param {string} options.logger - The logger.
-   *
-   * @returns {OAuth2Client} The `OAuth2Client` instance.
-   */
-  constructor({config, httpsAgent, issuer, logger} = {}) {
-    const cfg = config.find(a => a.issuer === issuer);
-    this.client_id = cfg.client_id;
-    this.client_secret = cfg.client_secret;
-    this.token_endpoint = cfg.token_endpoint;
-    this.grant_type = cfg.grant_type;
-    this.scope = cfg.scope;
-    this.maxRetries = cfg.maxRetries;
-    this.httpsAgent = httpsAgent;
-    this.logger = logger;
-  }
-
-  /**
+/**
    * Gets a new access token by first checking the database to see if an
    * access token already exists for the client id; if not it creates an
    * access token then stores it in the database.
    *
    * @param {object} options - Options to use.
-   * @param {string} options.retryAndExitOnFailure - If the request should be
-   *  retried indefinitely, and exit upon an unrecoverable error.
+   * @param {string} options.client_id - The ID of the client.
+   * @param {string} options.client_secret - The client secret.
+   * @param {string} options.token_endpoint - The URL to call.
+   * @param {string} options.grant_type - The grant type.
+   * @param {string} options.scope - The scope.
+   * @param {number} options.maxRetries - The maximum number of times to retry
+   *  the request.
+   *
    * @returns {Promise<string>} The access token.
    */
-  async getAccessToken({retryAndExitOnFailure = false}) {
-    const {
-      client_id, client_secret, token_endpoint, grant_type, scope, httpsAgent,
-      maxRetries
-    } = this;
-    if(retryAndExitOnFailure) {
-      return getAccessTokenRetryAndExitOnFailure({
-        client_id, client_secret, token_endpoint, grant_type, scope, httpsAgent
-      });
-    }
-    const fn = () => _getUncachedAccessToken({
-      client_id, client_secret, token_endpoint, grant_type, scope, httpsAgent,
-      maxRetries
-    });
-    return ACCESS_TOKEN_CACHE.memoize({key: client_id, fn});
-  }
+export async function getAccessToken({
+  client_id, client_secret, token_endpoint, grant_type, scope, maxRetries
+}) {
+  // it is ok to return the same cached token when the params `client_secret` or
+  // `maxRetries` are different, but these params below must be unique per token
+  const key = JSON.stringify([client_id, token_endpoint, grant_type, scope]);
```

Comment:
I think a comment here indicating an array is used here (instead of an object) to ensure that `JSON.stringify` preserves the order.  But it does beg the question, why not a string?  According to the jsdocs, all of these elements are strings themselves.
```js
const key = `${client_id}$(token_endpoint}$(grant_type}${scope}`;
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#issue_with_plain_json.stringify_for_use_as_javascript